packages/liquidator/src/liquidator.js

```diff
@@ -250,52 +273,122 @@ class Liquidator {
         parseInt(currentBlockTime) + this.liquidationDeadline
       );
 
-      // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      let liquidationResult, gasEstimation;
-      try {
-        [liquidationResult, gasEstimation] = await Promise.all([
-          liquidation.call({ from: this.account }),
-          liquidation.estimateGas({ from: this.account })
+      const syntheticTokenBalance = await this.syntheticToken.balanceOf(this.account);
+      const notEnoughTokens = syntheticTokenBalance.lt(tokensToLiquidate);
+
+      // Attempts to swap some capital for tokenCurrency
+      // Mutates the state of notEnoughCollateral in this logical scope
+      if (notEnoughTokens && this.oneInchClient !== undefined && this.oneInchClient !== null) {
```

Comment:
nit: do you. need to check both `undefined` and `null`? I think you can capture both by just replacing the last 2 conditionals with `!this.oneInchClient`. What do you think?
![Screen Shot 2020-08-27 at 07 47 02](https://user-images.githubusercontent.com/9457025/91438531-8166cf80-e839-11ea-9460-56c1e0cd9e44.png)